packages/framework/src/Model/Product/Brand/Brand.php

```diff
@@ -91,6 +102,75 @@ protected function createTranslation()
         return new BrandTranslation();
     }
 
+    /**
+     * @param \Shopsys\FrameworkBundle\Model\Product\Brand\BrandData $brandData
+     */
+    protected function setDomains(BrandData $brandData)
+    {
+        $domainIds = array_keys($brandData->seoTitles);
+        foreach ($domainIds as $domainId) {
+            try {
+                $brandDomain = $this->getBrandDomain($domainId);
+            } catch (BrandDomainNotFoundException $e) {
+                $brandDomain = new BrandDomain($this, $domainId);
+                $this->domains[$domainId] = $brandDomain;
+            }
+            $brandDomain->setSeoTitle($brandData->seoTitles[$domainId]);
+            $brandDomain->setSeoH1($brandData->seoH1s[$domainId]);
+            $brandDomain->setSeoMetaDescription($brandData->seoMetaDescriptions[$domainId]);
+        }
+    }
+
+    /**
+     * @return \Shopsys\FrameworkBundle\Model\Product\Brand\BrandDomain[]|\Doctrine\Common\Collections\ArrayCollection
+     */
+    public function getDomains()
+    {
+        return $this->domains;
+    }
+
+    /**
+     * @param int $domainId
+     * @return \Shopsys\FrameworkBundle\Model\Product\Brand\BrandDomain
+     */
+    protected function getBrandDomain(int $domainId)
+    {
+        foreach ($this->domains as $domain) {
```

Comment:
The ProductDomains are not necessarily indexed by domain ID (and it's probably unwise to index it in such a way in https://github.com/shopsys/shopsys/pull/165/files/0ff35b922475cab1d02d6b2ce9a3b452d4906834#diff-ce1e32a3065400b38b9b2cac44e743e3R116). That's because Doctrine will hydrate the property as a unordered collection - it won't even keep the order of the array, let alone the indexes. Indexes are not persisted in any way by the Doctrine.